### PR TITLE
Field Notice 0039 CLI Update

### PR DESCRIPTION
**Pull request by andriirr**
This to update CLI in EOS images starting from version 4.21.3 It also includes FN39 CLI changes.

EOS 4.21.3

OLD|NEW

vrrp delay reload | vrrp timers delay reload
vrrp description | vrrp session description
vrrp ip | vrrp ipv4
vrrp ip secondary | vrrp ipv4 secondary
vrrp priority | vrrp priority-level
vrrp shutdown | vrrp disabled
vrrp timers advertise | vrrp advertisement interval
vrrp track | vrrp tracked-object

EOS 4.23

OLD|NEW

neighbor peer-group | neighbor peer group
ntp source | ntp local-interface
username sshkey | username ssh-key
vrf definition | vrf instance
vrf forwarding | vrf

### CODE REVIEW

File: test/unit/test_client.py [hunk 1]
```diff
@@ -233,18 +233,18 @@ def test_load_config_for_connection_with_filename(self):
         conf = get_fixture('eapi.conf')
         pyeapi.client.load_config(filename=conf)
         cfg = pyeapi.client.config.get_connection('test1')
-        self.assertEqual(cfg['host'], '192.168.1.16')
-        self.assertEqual(cfg['username'], 'eapi')
-        self.assertEqual(cfg['password'], 'password')
+        self.assertEqual(cfg['host'], '10.16.129.149')
+        self.assertEqual(cfg['username'], 'arista')
+        self.assertEqual(cfg['password'], 'arista')
```
Comment (mharista) [inline test/unit/test_client.py]: Do we need these fields updated? I assume 10.16.129.149 is your dev environment.

File: test/unit/test_client.py [hunk 2]
```diff
@@ -233,18 +233,18 @@ def test_load_config_for_connection_with_filename(self):
...
     def test_load_config_for_connection_with_env(self):
         os.environ['EAPI_CONF'] = get_fixture('eapi.conf')
         pyeapi.client.load_config(random_string())
         cfg = pyeapi.client.config.get_connection('test1')
-        self.assertEqual(cfg['host'], '192.168.1.16')
-        self.assertEqual(cfg['username'], 'eapi')
-        self.assertEqual(cfg['password'], 'password')
+        self.assertEqual(cfg['host'], '10.16.129.149')
+        self.assertEqual(cfg['username'], 'arista')
+        self.assertEqual(cfg['password'], 'arista')
```
Comment (mharista) [inline test/unit/test_client.py]: Same comment as above.

File: test/unit/test_client.py [hunk 3]
```diff
@@ -418,11 +418,11 @@ def test_connect_return_node(self):
         with patch.dict(pyeapi.client.TRANSPORTS, {'https': transport}):
             conf = get_fixture('eapi.conf')
             pyeapi.client.load_config(filename=conf)
-            node = pyeapi.client.connect(host='192.168.1.16', username='eapi',
-                                         password='password', port=None,
+            node = pyeapi.client.connect(host='10.16.129.149', username='arista',
+                                         password='arista', port=None,
```
Comment (mharista) [inline test/unit/test_client.py]: Same as above

File: test/unit/test_client.py [hunk 4]
```diff
@@ -418,11 +418,11 @@ def test_connect_return_node(self):
...
                                          timeout=60, return_node=True)
-            kwargs = dict(host='192.168.1.16', username='eapi',
-                          password='password', port=None, key_file=None,
+            kwargs = dict(host='10.16.129.149', username='arista',
+                          password='arista', port=None, key_file=None,
```
Comment (mharista) [inline test/unit/test_client.py]: Same as above.

File: test/unit/test_client.py [hunk 5]
```diff
@@ -432,12 +432,12 @@ def test_connect_return_node_enablepwd(self):
         with patch.dict(pyeapi.client.TRANSPORTS, {'https': transport}):
             conf = get_fixture('eapi.conf')
             pyeapi.client.load_config(filename=conf)
-            node = pyeapi.client.connect(host='192.168.1.16', username='eapi',
-                                         password='password', port=None,
+            node = pyeapi.client.connect(host='10.16.129.149', username='arista',
+                                         password='arista', port=None,
```
Comment (mharista) [inline test/unit/test_client.py]: Same as above

File: test/unit/test_client.py [hunk 6]
```diff
@@ -432,12 +432,12 @@ def test_connect_return_node_enablepwd(self):
...
                                          timeout=60, enablepwd='enablepwd',
                                          return_node=True)
-            kwargs = dict(host='192.168.1.16', username='eapi',
-                          password='password', port=None, key_file=None,
+            kwargs = dict(host='10.16.129.149', username='arista',
+                          password='arista', port=None, key_file=None,
```
Comment (mharista) [inline test/unit/test_client.py]: Same as above.

File: test/unit/test_client.py [hunk 7]
```diff
@@ -448,8 +448,8 @@ def test_connect_to_with_config(self):
             conf = get_fixture('eapi.conf')
             pyeapi.client.load_config(filename=conf)
             node = pyeapi.client.connect_to('test1')
-            kwargs = dict(host='192.168.1.16', username='eapi',
-                          password='password', port=None, key_file=None,
+            kwargs = dict(host='10.16.129.149', username='arista',
+                          password='arista', port=None, key_file=None,
```
Comment (mharista) [inline test/unit/test_client.py]: Same as above.

File: pyeapi/client.py [hunk 8]
```diff
@@ -516,6 +517,14 @@ def version_number(self):
...
         return self._version_number
 
+    @property
+    def version_id(self):
+        if self._version_id:
+            return self._version_id
+        output = self.enable('show version')
+        self._version_id = output[0]['result']['version'].split()[0][:-1]
+        return self._version_id
+
```
[the remark this answers is not in the document]
Comment (andriirr) [reply pyeapi/client.py]: Have adjusted version_number to be used for version checks as well. Removed version_id.

File: test/system/test_client.py [hunk 9]
```diff
@@ -57,7 +58,10 @@ def setUp(self):
                 if dut._enablepwd is not None:
                     # If enable password defined for dut, set the
                     # enable password on the dut and clear it on tearDown
-                    dut.config("enable secret %s" % dut._enablepwd)
+                    if dut.version_number >= '4.23':
```
[the remark this answers is not in the document]
Comment (mharista) [reply test/system/test_client.py]: From what I can tell version_id checks are changed to version_number.

**Comment by mharista** [comment]
Looks good to me.

File: test/system/test_client.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (andriirr) [reply test/system/test_client.py]: yup, have reused existing func

**Pull request merged**